src/lib.rs

```diff
@@ -43,6 +43,16 @@ pub struct ToipeError {
     pub msg: String,
 }
 
+impl ToipeError {
+    /// Prefixes a ToipeError's msg with 'msg_context'
+    ///
+    /// The ToipeError operated on is consumed, and a new one is returned
```

Comment:
Nitpick: This is evident from the function signature, no need to mention it here.